File: Programming-languages-and-laboratory/Python/Exercises/21a_compito_java_20180406/GestioneCatenaRistoranti.py
from Ristorante import Ristorante
from Pub import Pub
from Osteria import Osteria
from Menu import Menu


class GestioneCatenaRistoranti:
    def main():

        ristoranti = []
        cod_to_rist = {}

        try:
            f = open("ristoranti.txt", "r")
            line = f.readline().strip()

            while line != "":
                tok = line.split()
                codice = int(tok[0])
                tipo = tok[1]
                nome_ristorante = f.readline().strip()
                line = f.readline().strip()
                tok = line.split()

                if tipo == "pub":
                    dipendenti = int(tok[0])
                    tavoli = int(tok[1])

                    pu = Pub(codice, tipo, nome_ristorante, dipendenti, tavoli)
                    ristoranti.append(pu)
                    cod_to_rist[codice] = pu

                else:
                    posti = int(tok[0])
                    superficie = float(tok[1])
                    bagno_disabili = tok[2]

                    ost = Osteria(codice, tipo, nome_ristorante,
                                  posti, superficie, bagno_disabili)
                    ristoranti.append(ost)
                    cod_to_rist[codice] = ost

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        print("nome, codice, tipo, n. dipendenti, n. tavoli, n. posti, superficie, bagno disabili")

        for r in ristoranti:
            print(r)
        print()

        try:
            f = open("menu.txt", "r")
            line = f.readline().strip()

            while line != "":
                codice_ristorante = int(line)
                descrizione_prodotto = f.readline().strip()
                line = f.readline().strip()
                tok = line.split()
                tipo = tok[0]
                prezzo = float(tok[1])
                costo_materie_prime = float(tok[2])
                line = f.readline().strip()

                me = Menu(codice_ristorante, descrizione_prodotto,
                          tipo, prezzo, costo_materie_prime)
                cod_to_rist[codice_ristorante].add_menu(me)
                cod_to_rist[codice_ristorante].add_prezzo_totale(prezzo)

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        for ri in ristoranti:
            print(ri.stampa_menu())
        print()

        nome_ristorante_costoso = ""
        prezzo_medio_max = 0.0

        for ris in ristoranti:
            if ris.get_prezzo_medio() > prezzo_medio_max:
                prezzo_medio_max = ris.get_prezzo_medio()
                nome_ristorante_costoso = ris.get_nome_ristorante()

        print("Il locale con il prezzo medio più alto è " +
              nome_ristorante_costoso + ".")


if __name__ == "__main__":
    GestioneCatenaRistoranti.main()

File: II - Linguaggi di programmazione e laboratorio/Esame/Java/21_compito_java_2018_04_06a/GestioneCatenaRistoranti.java
import java.util.*;
import java.io.*;

public class GestioneCatenaRistoranti {
	static List<Ristorante> ristoranti = new LinkedList<Ristorante>();

	public static void main(String[] args) {

		// Punto 1, memorizzazione ristoranti

		try {
			BufferedReader br = new BufferedReader(new FileReader("ristoranti.txt"));
			String line = br.readLine();

			while (line != null) {
				StringTokenizer tok = new StringTokenizer(line);
				int codiceRistorante = Integer.parseInt(tok.nextToken());
				String tipoRistorante = tok.nextToken();
				String nomeRistorante = br.readLine();

				line = br.readLine();

				if (tipoRistorante.equals("pub")) {
					tok = new StringTokenizer(line);
					int numeroDipendenti = Integer.parseInt(tok.nextToken());
					int numeroTavoli = Integer.parseInt(tok.nextToken());

					Ristorante r = new Pub(codiceRistorante, tipoRistorante, nomeRistorante, numeroDipendenti,
							numeroTavoli);
					ristoranti.add(r);

				} else {
					tok = new StringTokenizer(line);
					int postiASedere = Integer.parseInt(tok.nextToken());
					double superficie = Double.parseDouble(tok.nextToken());
					boolean bagnoDisabili = Boolean.parseBoolean(tok.nextToken());

					Ristorante r = new Osteria(codiceRistorante, tipoRistorante, nomeRistorante, postiASedere,
							superficie, bagnoDisabili);
					ristoranti.add(r);

				}

				line = br.readLine();

			}

			br.close();

		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}

		// Punto 2, memorizzazione menu

		try {
			BufferedReader br = new BufferedReader(new FileReader("menu.txt"));
			String line = br.readLine();

			while (line != null) {
				int codiceRistorante = Integer.parseInt(line);
				String descrizioneProdotto = br.readLine();
				line = br.readLine();
				StringTokenizer tok = new StringTokenizer(line);
				String tipoProdotto = tok.nextToken();
				float prezzoProdotto = Float.parseFloat(tok.nextToken());
				float costoMateriePrime = Float.parseFloat(tok.nextToken());
				line = br.readLine();
				line = br.readLine();

				Menu m = new Menu(codiceRistorante, descrizioneProdotto, tipoProdotto, prezzoProdotto,
						costoMateriePrime);

				for (Ristorante r : ristoranti) {
					if (r.codiceRistorante == m.codiceRistorante) {
						r.menu.add(m);
					}
				}

			}

			br.close();

		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}

		// Punto 3, stampa tutti i ristoranti - 6 punti

		System.out.println("nome\tcodice\ttipo\tn. dipendenti\tn. tavoli\tn. posti\tsuperficie\tbagno disabili");

		for (Ristorante r : ristoranti) {
			System.out.println(r);
		}

		System.out.println("");

		// Punto 4, stampa locali con nome e menu

		for (Ristorante ris : ristoranti) {
			System.out.println(ris.nomeRistorante + " " + ris.menu);
		}

		// Punto 5, stampa locale con prezzo medio dei prodotti piu alto

		float max = 0;
		String nome = null;
		for (Ristorante l : ristoranti) {
			float media = l.getMedia();
			if (media > max) {
				max = media;
				nome = l.nomeRistorante;
			}
		}
		System.out.println("\nIl ristorante con il prezzo medio pi alto è " + nome + ".");

	}

}
